lib/models/message.ts

```diff
@@ -301,4 +306,65 @@ export class MessageModel extends DocumentDbModel<
 
     return right(errorOrAttachmentMeta.value);
   }
+
+  /**
+   * Get stored MessageContent (if any) from blob service.
+   */
+  public async getStoredContent(
+    blobService: BlobService,
+    messageId: string,
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, Option<MessageContent>>> {
+    // get link to attached blob(s)
+    const media = await iteratorToArray(
+      await this.getAttachments(messageId, {
+        partitionKey: fiscalCode
+      })
+    );
+
+    // no blob(s) attached to the message
+    if (!media || !media[0]) {
+      return right<Error, Option<MessageContent>>(none);
+    }
+
+    const blobId = blobIdFromMessageId(messageId);
+
+    // retrieve blob content and deserialize
+    const maybeContentAsTextOrError = await getBlobAsText(
+      blobService,
+      this.containerName,
+      blobId
+    );
+
+    if (isLeft(maybeContentAsTextOrError)) {
+      return left<Error, Option<MessageContent>>(
+        maybeContentAsTextOrError.value
+      );
+    }
+
+    // media exists but the content is empty
+    const maybeContentAsText = maybeContentAsTextOrError.value;
+    if (isNone(maybeContentAsText)) {
+      return left<Error, Option<MessageContent>>(
+        new Error("Cannot get stored message content from attachment")
```

Comment:
this is actually "message attachment is empty" ?

Reply:
no, when the content is empty result = some("").

this could never happen in theory as the azure sdk api call should always returns a non null/undefined result.